parsec/api/protocol/block.py

```diff
@@ -1,100 +1,42 @@
 # Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
 
-from typing import TYPE_CHECKING, Union, Type
-import attr
+from typing import Type, Any
 
 from parsec.serde import fields
-from parsec.api.protocol.base import (
-    BaseReqSchema,
-    BaseRepSchema,
-    BaseTypedReqSchema,
-    BaseTypedRepSchema,
-    CmdSerializer,
-    BaseReq,
-    BaseRep,
-    cmd_rep_error_type_factory,
-    cmd_rep_factory,
-)
-from parsec.api.protocol.realm import RealmIDField
 from parsec._parsec import BlockID
 
 __all__ = ("BlockID", "BlockIDField", "block_create_serializer", "block_read_serializer")
 
 
 BlockIDField: Type[fields.Field] = fields.uuid_based_field_factory(BlockID)
 
+from parsec._parsec import (
+    BlockReadReq,
+    BlockReadRep,
+    BlockCreateReq,
+    BlockCreateRep,
+)
 
-class BlockCreateReqSchema(BaseReqSchema):
-    block_id = BlockIDField(required=True)
-    realm_id = RealmIDField(required=True)
-    block = fields.Bytes(required=True)
-
-
-class BlockCreateRepSchema(BaseRepSchema):
-    pass
-
-
-block_create_serializer = CmdSerializer(BlockCreateReqSchema, BlockCreateRepSchema)
-
-
-@attr.s(slots=True, frozen=True, auto_attribs=True, kw_only=True, eq=False)
-class BlockReadReq(BaseReq):
-    class SCHEMA_CLS(BaseTypedReqSchema):
-        cmd = fields.CheckedConstant("block_read", required=True)
-        block_id = BlockIDField(required=True)
-
-    block_id: BlockID
-
-
-@attr.s(slots=True, frozen=True, auto_attribs=True, kw_only=True, eq=False)
-class BlockReadRepOk(BaseRep):
-    class SCHEMA_CLS(BaseTypedRepSchema):
-        status = fields.CheckedConstant("ok", required=True)
-        block = fields.Bytes(required=True)
-
-    block: bytes
 
+class BlockCreateSerializer:
+    @staticmethod
+    def req_dumps(req: dict[str, Any]) -> bytes:
+        return BlockCreateReq(req["block_id"], req["realm_id"], req["block"]).dump()
```

Comment:
use `BlockCreateReq(**req)` instead

this will fail if the wrong arguments are provided, and it's generic over all the commands so you'll thank more for the next commands ^^

Reply:
And given the serializer is all generic with this trick, we can have a common `ApiCommandSerializer` class for all commands:
```python
class ApiCommandSerializer:
    def __init__(self, req_schema, rep_schema):
        self.req_schema = req_schema
        self.rep_schema = rep_schema
    def req_dumps(self, req: dict[str, Any]) -> bytes:
        return self.req_schema(**req).dump()
    def rep_loads(self, raw: bytes) -> Any:
        return self.rep_schema.load(raw)

block_create_serializer = ApiCommandSerializer(BlockCreateReq, BlockCreateRep)
```